# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../luaexpat/lxplib-no-compat-5.1.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- luaexpat-1.0.1-vanilla/src/lxplib.c	2005-03-28 19:53:22.000000000 +0200
+++ luaexpat-1.0.1/src/lxplib.c	2006-03-13 21:16:55.000000000 +0100
@@ -13,7 +13,6 @@
 
 #include "lua.h"
 #include "lauxlib.h"
-#include "compat-5.1.h"
 
 
 #include "lxplib.h"
@@ -335,7 +334,7 @@
   while (lua_next(L, 1)) {
     lua_pop(L, 1);  /* remove value */
     if (lua_type(L, -1) != LUA_TSTRING ||
-        luaL_findstring(lua_tostring(L, -1), validkeys) < 0)
+        luaL_checkoption(L, -1, NULL, validkeys) < 0)
       luaL_error(L, "invalid key `%s' in callback table", lua_tostring(L, -1));
   }
 }
